mllib/src/main/scala/org/apache/spark/ml/functions.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml
+
+import org.apache.spark.annotation.Since
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.mllib.linalg.{Vector => OldVector}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.udf
+
+// scalastyle:off
+@Since("3.0.0")
+object functions {
+// scalastyle:on
+
+  private[ml] val vector_to_array_udf = udf { vec: Any =>
```

Comment:
UDF itself isn't an efficient way to do it as it requires to ser/de for every row.

Reply:
Seems right. But directly transform on raw binary data make implementation complicated. Should we do that ? @mengxr

Reply:
And note that in MLlib transform/predict, we already use UDF widely...

Reply:
Okay, I am fine if https://github.com/apache/spark/pull/26910#issuecomment-566842696 doesn't work. Maybe it's because I am not used to ML/MLlib.